spec/02-integration/22-ai_plugins/01-reports_spec.lua

```diff
@@ -0,0 +1,185 @@
+                      end
+                    }
+                  }
+              }
+            ]]
+
+      local http_srv = assert(bp.services:insert {
+        name = "mock-service",
+        host = helpers.mock_upstream_host,
+        port = helpers.mock_upstream_port,
+      })
+
+      local chat_good = assert(bp.routes:insert {
+        service = http_srv,
+        protocols = { "http" },
+        hosts = { "http-service.test" }
+      })
+
+      bp.plugins:insert({
+        name = "reports-api",
+        config = {}
+      })
+
+      bp.plugins:insert {
+        name = PLUGIN_NAME,
+        id = "6e7c40f6-ce96-48e4-a366-d109c169e444",
+        route = { id = chat_good.id },
+        config = {
+          route_type = "llm/v1/chat",
+          logging = {
+            log_payloads = false,
+            log_statistics = true,
+          },
+          auth = {
+            header_name = "Authorization",
+            header_value = "Bearer openai-key",
+          },
+          model = {
+            name = "gpt-3.5-turbo",
+            provider = "openai",
+            options = {
+              max_tokens = 256,
+              temperature = 1.0,
+              upstream_url = "http://"..helpers.mock_upstream_host..":"..MOCK_PORT.."/llm/v1/chat/good"
+            },
+          },
+        },
+      }
+
+      assert(helpers.start_kong({
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+        database = strategy,
+        dns_hostsfile = dns_hostsfile,
+        plugins = "bundled,reports-api",
+        anonymous_reports = true,
+      }, nil, nil, fixtures))
+
+    end)
+
+    lazy_teardown(function()
+      os.remove(dns_hostsfile)
+
+      helpers.stop_kong()
+    end)
+
+    before_each(function()
+      reports_server = helpers.tcp_server(constants.REPORTS.STATS_TLS_PORT, {tls=true})
+    end)
+
+    describe("check report has ai data", function()
+      it("logs number of requests triggering a ai plugin", function()
+        local proxy_client = assert(helpers.proxy_client())
+        local res = proxy_client:get("/", {
+          headers = { 
+            ["content-type"] = "application/json",
+            ["accept"] = "application/json",
+            ["host"]  = "http-service.test",
+          },
+          body = pl_file.read("spec/fixtures/ai-proxy/openai/llm-v1-chat/requests/good.json"),
+        })
+        assert.res_status(200, res)
+
+        local proxy_client2 = assert(helpers.proxy_client())
+        local res2 = proxy_client2:get("/", {
+          headers = { 
+            ["content-type"] = "application/json",
+            ["accept"] = "application/json",
+            ["host"]  = "http-service.test",
+          },
+          body = pl_file.read("spec/fixtures/ai-proxy/openai/llm-v1-chat/requests/good.json"),
+        })
+        assert.res_status(200, res2)
+
+        reports_send_ping(constants.REPORTS.STATS_TLS_PORT)
+
+        proxy_client:close()
+        proxy_client2:close()
+
+        ngx.sleep(0.1)
```

Comment:
`sleep` is an anti-pattern in tests.  Consider using `helpers.eventually`.

Reply:
Thanks @hanshuebner 
agree and we actually don't need the sleep function so I removed it.

Do notice that the test will fail until https://github.com/Kong/kong/pull/12903 is merged. 

Thanks